### PR TITLE
SLING-7759 add new patterns to filter annotation

### PR DESCRIPTION
**Pull request by npeltier**
- moved scope first,
- added new restrictions to the annotation

### CODE REVIEW

**Comment by npeltier** [comment]
sorry about the messed up diff for javadoc, not sure why this happened

File: src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java [hunk 1]
```diff
@@ -43,21 +43,52 @@
...
-     * @return the scope of the filter
+    /**
+     * Restrict the filter to paths that match the supplied regular expression. Requires Sling Engine 2.4.0.
+     * @return the pattern to restrict the filter
      */
-    SlingServletFilterScope[] scope() default SlingServletFilterScope.REQUEST;
+    String suffix_pattern() default "";
+
+    /**
+     * Restrict the filter to resources that match any of the the supplied resource types. Requires Sling Engine 2.7.0.
```
Comment (kwin) [inline src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java]: Can you clarify whether absolute or relative resource types should be given here?

File: src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java [hunk 2]
```diff
@@ -43,21 +43,52 @@
...
+
+    /**
+     * Restrict the filter to methods that match request's. Requires Sling Engine 2.7.0.
+     * @return the methods to restrict the filter
+     */
+    String[] methods() default {};
+
+    /**
+     * Restrict the filter to extensions that match request's. Requires Sling Engine 2.7.0.
+     * @return the methods to restrict the filter
```
Comment (kwin) [inline src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java]: methods -> extensions

File: src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java [hunk 3]
```diff
@@ -43,21 +43,52 @@
...
+     */
+    String[] extensions() default {};
+
+    /**
+     * Restrict the filter to selectors that match any of the the request's. Requires Sling Engine 2.7.0.
```
Comment (kwin) [inline src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java]: the the

File: src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java [hunk 4]
```diff
@@ -43,21 +43,52 @@
...
+     * @return the resourceTypes to restrict the filter
+     */
```
Comment (kwin) [inline src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java]: resourceTypes->selectors

File: src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java (hunk 3 above)
Comment (kwin) [reply src/main/java/org/apache/sling/servlets/annotations/SlingServletFilter.java]: What happens if the request has multiple selectors? How are those exactly matched?

**Review by kwin — COMMENTED** [review]
Please clarify for all elements, what happens if they are left empty.

**Comment by npeltier** [comment]
thx for the review, @kwin! I think i've addressed all your comments

**Comment by npeltier** [comment]
@kwin are we good?

**Review by kwin — APPROVED** [review]

**Comment by kwin** [comment]
Looks good to me now, thanks!

**Pull request merged**